packages/openapi3-parser/lib/parser/oas/parseSchemaObject.js

```diff
@@ -94,16 +118,14 @@ function parseType(context) {
   }
 
   const ensureValidType = R.unless(
-    isValidType,
-    R.compose(
-      createWarning(context.namespace, `'${name}' 'type' must be either ${types.join(', ')}`),
-      getValue
-    )
+    element => isValidType(element.toValue()),
+    createWarning(context.namespace, `'${name}' 'type' must be either ${types.join(', ')}`)
   );
 
   return pipeParseResult(context.namespace,
-    parseString(context, name, false),
-    ensureValidType);
+    R.unless(isString, value => createWarning(context.namespace, `'${name}' 'type' is not a string`, value)),
```

Comment:
Eventually this branch will head into a direction where we handle multiple types, I'd imagine it will use `R.cond` such as:

```js
R.cond([
  [isString, ensureValidType, then cocerce to array],
  [isArray, ...],
  [R.T, createWarning(value must be ...)],
])
```